### PR TITLE
mesh-init: Configure Envoy to talk over TLS to the Consul client

### PR DESCRIPTION
**Pull request by ishustava**
This PR proposes the following changes:
- Change mesh-init command to configure Envoy to use TLS with a trusted CA cert. We're using a trusted CA here because currently Consul doesn't allow you to generate Envoy bootstrap json with TLS verification skipped, and we didn't want to modify the Envoy json ourselves.
- Add `dev-docker` to `Makefile`
- fix dockerignore to exclude the right binary so that building a docker image locally works.

### CODE REVIEW

**Review by pglass — APPROVED** [review]
Looks good to me! Tests passed locally.

Couple of nit picks - not important.

File: subcommand/mesh-init/command_test.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+			}))
+			t.Cleanup(ecsMetadataServer.Close)
+			os.Setenv(awsutil.ECSMetadataURIEnvVar, ecsMetadataServer.URL)
+
+			ui := cli.NewMockUi()
+			cmd := Command{
+				UI: ui,
+			}
+
+			envoyBootstrapFile, err := ioutil.TempFile("", "")
```
[the remark this answers is not in the document]
Comment (ishustava) [reply subcommand/mesh-init/command_test.go:103]: Great catch! will update it!

File: subcommand/mesh-init/command_test.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+			expectedServiceRegistration := &api.AgentService{
+				ID:      "test-service-abcdef",
+				Service: "test-service",
+				Port:    c.servicePort,
+				Meta: map[string]string{
+					"task-id":  "abcdef",
+					"task-arn": "arn:aws:ecs:us-east-1:123456789:task/test/abcdef",
+					"source":   "consul-ecs",
+				},
+			}
```
[the remark this answers is not in the document]
Comment (ishustava) [reply subcommand/mesh-init/command_test.go:128]: yep, good idea!

File: subcommand/mesh-init/command.go [hunk 3]
```diff
@@ -154,14 +158,57 @@ func (c *Command) realRun(log hclog.Logger) error {
 
 	log.Info("service and proxy registered successfully", "name", serviceName, "id", serviceID)
 
+	// Grab CA root from the Consul server if TLS is enabled.
```
Comment (lkysow) [inline subcommand/mesh-init/command.go:161]: Why not disable CA verification for Envoy since it's talking localhost to Consul so there's no MITM risk.

Comment (ishustava) [reply subcommand/mesh-init/command.go:161]: How do you disable TLS verification for envoy through the Consul cli? I haven't found anything in the consul codebase that would read `CONSUL_HTTP_SSL_VERIFY` env var for the `consul connect envoy -bootstrap` command or a flag that would do similar for that command.

Comment (lkysow) [reply subcommand/mesh-init/command.go:161]: So for the initial call that that command makes itself to Consul, I think it respects the env var. The problem is that the generated bootstrap config doesn't have verification turned off. I edited the generated config in my branch to work around this: https://github.com/hashicorp/consul-ecs/blob/secure/subcommand/mesh-init/command.go#L209-L226

**Review by lkysow — APPROVED** [review]
🚀

**Comment by ishustava** [comment]
Closing this PR as we've decided to not have TLS for HTTP and gRPC for the consul client.

I've moved the unit tests and other fixes to #18.

**Pull request closed without merging**